zebra/kernel_netlink.c

```diff
@@ -522,7 +522,7 @@ netlink_parse_info (int (*filter) (struct sockaddr_nl *, struct nlmsghdr *,
       if (IS_ZEBRA_DEBUG_KERNEL_MSGDUMP_RECV)
         {
           zlog_debug("%s: << netlink message dump [recv]", __func__);
-          zlog_hexdump(&msg, sizeof(msg));
+          zlog_hexdump(buf, status);
```

Comment:
forget my comment, I didn't notice that we are still in the while loop and those two error cases would break out of the loop. status  is guaranteed to have a valid size value by the time we get to call.